apps/backoffice-v2/src/pages/Entity/components/EditableDetails/EditableDetails.tsx

```diff
@@ -59,46 +59,43 @@ export const EditableDetails: FunctionComponent<IEditableDetails> = ({
   const form = useForm({
     defaultValues,
   });
-  const { mutate: mutateUpdateWorkflowById } = useUpdateWorkflowByIdMutation({
+  const { mutate: mutateUpdateWorkflowById } = useUpdateDocumentByIdMutation({
     workflowId,
+    documentId: valueId,
   });
   const onMutateTaskDecisionById = ({
-    context,
+    // Ask Omri about this
+    document,
     action,
   }: {
-    context: AnyRecord;
+    document: AnyRecord;
     action: Parameters<typeof mutateUpdateWorkflowById>[0]['action'];
   }) =>
     mutateUpdateWorkflowById({
-      context,
+      document,
       action,
     });
   const onSubmit: SubmitHandler<Record<PropertyKey, unknown>> = formData => {
-    const context = {
-      documents: documents?.map(document => {
-        if (document?.id !== valueId) return document;
+    const document = documents?.find(document => document?.id === valueId);
```

Comment:
Do we need to use the entire document here over just type, category, and properties?